src/promptflow/promptflow/_sdk/_submitter/run_submitter.py

```diff
@@ -63,8 +64,12 @@ def _run_bulk(self, run: Run, stream=False, **kwargs):
 
     def _submit_bulk_run(self, flow: Flow, run: Run, local_storage: LocalStorageOperations) -> dict:
         run_id = run.name
-        with _change_working_dir(flow.code):
-            connections = SubmitterHelper.resolve_connections(flow=flow)
+        if flow.dag.get("language", FlowLanguage.Python) == FlowLanguage.CSharp:
+            BatchEngine.register_executor(FlowLanguage.CSharp, CSharpExecutorProxy)
```

Comment:
I think we may register this in the initialization phase.

Reply:
Updated